system/jlib/jfile.cpp

```diff
@@ -3264,7 +3264,8 @@ void doCopyFile(IFile * target, IFile * source, size32_t buffersize, ICopyFilePr
         // try to delete partial copy
         StringBuffer s;
         s.append("copyFile target=").append(dest->queryFilename()).append(" source=").append(source->queryFilename()).appendf("; read/write failure (%d): ",e->errorCode());
-        exc.setown(MakeStringException(e->errorCode(), "%s", s.str()));
+        e->errorMessage(s);
```

Comment:
I think this is probably right. In this case e  will be a IOSException or an IErrnoException.   (I think they are only different for windows - the first for operating system errors, the second for errors from the c library first...!)

When e->errorMessage() is called it will call strerror() - so the message will include the description.  I think it cannot call makeErrnoException() because it may not be an errno.

There is similar code in copyFileSection() ~995 which adds the error message and creates an OsException - which I think will lead to the message being duplicated, but I didn't fix since it would be benign.

Reply:
ok, then looks fine.